File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/ProductPriceDto.cs
using ProjectPos.Services.EntityDtos;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectPos.Services.DTOs
{
    public class ProductPriceDto : FullAuditedEntityDto<int>
    {
        public int? ProductInventoryId { get; set; }
        public string? BarCode { get; set; }
        public string? Name { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Cost { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? MarkUp { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Price { get; set; }
        public string? Notes { get; set; }

        [ForeignKey("ProductInventoryId")]
        public ProductInventoryDto? Product { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/product-price-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { ProductInventoryDto } from "./product-inventory-dto";

export interface ProductPriceDto extends FullAuditedEntityDto<number> {
    productInventoryId?: number;
    name?: string;
    barCode?: number;
    cost?: number;
    markUp?: number;
    price?: number;
    product?: ProductInventoryDto;
}
